x-pack/plugins/cases/public/components/files/files_table.tsx

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useMemo, useState } from 'react';
+
+import type { EuiBasicTableColumn, Pagination, EuiBasicTableProps } from '@elastic/eui';
+import type { FileJSON } from '@kbn/shared-ux-file-types';
+
+import {
+  EuiModalBody,
+  EuiModalFooter,
+  EuiButtonIcon,
+  EuiLink,
+  EuiBasicTable,
+  EuiLoadingContent,
+  EuiSpacer,
+  EuiText,
+  EuiEmptyPrompt,
+  EuiButton,
+  EuiModal,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+} from '@elastic/eui';
+import { useFilesContext } from '@kbn/shared-ux-file-context';
+import { FileImage } from '@kbn/shared-ux-file-image';
+
+import * as i18n from './translations';
+import { APP_ID } from '../../../common';
+import { CASES_FILE_KINDS } from '../../files';
+
+const EmptyFilesTable = () => (
+  <EuiEmptyPrompt
+    title={<h3>{'No attachments available'}</h3>}
+    data-test-subj="files-table-empty"
+    titleSize="xs"
+    actions={
+      <EuiButton
+        size="s"
+        iconType="plusInCircle"
+        data-test-subj="case-detail-files-table-upload-file"
+      >
+        {i18n.ADD_FILE}
+      </EuiButton>
+    }
+  />
+);
+
+EmptyFilesTable.displayName = 'EmptyFilesTable';
+
+interface FilesTableProps {
+  isLoading: boolean;
+  items: FileJSON[];
+  onChange: EuiBasicTableProps<FileJSON>['onChange'];
+  pagination: Pagination;
+}
+
+export const FilesTable = ({ items, pagination, onChange, isLoading }: FilesTableProps) => {
```

Comment:
Should we memoize this component with `React.memo`?

Reply:
Hmm, the `FilesTable` is the top-level component, do you mean maybe the `columns`?

Reply:
The `columns` were moved to a separate file btw

Reply:
> Hmm, the FilesTable is the top-level component, do you mean maybe the columns?

Na, I was referring to the `FilesTable`. I think we could still memoize it right?

I don't know if we'd get a performance gain from memoizing this component or not though, I guess it depends if it renders a lot with the same file `items`?

For heavy components we do this I think: https://github.com/elastic/kibana/blob/main/x-pack/plugins/cases/public/components/all_cases/all_cases_list.tsx#L70

Reply:
I didn't notice we had it for the all cases list 😄 

Isn't this better applicable only when the component to be rendered does not change that much? These tables have filtering options, sorting options, pagination....

Visualizing them varies so much that I am not sure there is any point in memoizing.

Reply:
> Isn't this better applicable only when the component to be rendered does not change that much? These tables have filtering options, sorting options, pagination....

Yeah seems reasonable to me 👍